standard/power_plant.go

```diff
@@ -0,0 +1,75 @@
+package standard
+
+import (
+	"github.com/hinshun/powergrid"
+)
+
+// PowerPlant is an implementation of the powergrid.PowerPlant
+// interface that works for the game's standard power plants
+type PowerPlant struct {
+	ordinal       uint
+	fuelType      []powergrid.Resource
+	fuelRequired  uint
+	powerCapacity uint
+	inventory     powergrid.ResourceSet
+}
+
+// NewPowerPlant constructs a standard PowerPlant
+func NewPowerPlant(
+	ordinal uint,
+	fuelType []powergrid.Resource,
+	fuelRequired uint,
+	powerCapacity uint) *PowerPlant {
+	return &PowerPlant{
+		ordinal:       ordinal,
+		fuelType:      fuelType,
+		fuelRequired:  fuelRequired,
+		powerCapacity: powerCapacity,
+		inventory:     make(powergrid.ResourceSet),
+	}
+}
+
+// Ordinal returns the ordinal of the PowerPlant
+func (plant *PowerPlant) Ordinal() uint {
```

Comment:
The go pattern is to use shorten the self reference to the capital letters of your struct, in this case: `pp`